### DISCUSSION TITLE
where is setup/network invoked?

### DISCUSSION THREAD

**Discussion by tve** [Moddable-OpenSource/moddable#1161] [Q&A] [answered]
The code to initialize Wifi is in `build/devices/<target>/setup/network.js` and I see that the module is defined and preloaded as "setup/network" in `examples/manifest_net.json`, and that is pulled into any app that pulls in `modules/io/manifest.json` (which I find somewhat unfortunate).
Now setup/network exports a default function but I don't see any `import ... from "setup/network"` anywhere. Where is that default function invoked and why? (xsbug just says "host" in the call stack)

I'd like to init Wifi differently, is there a way to exclude setup/network without essentially copying large portions of the manifest.jsons? And is there a way to get my module to be called as early as setup/network seems to be called? (I have mine set to preload and it also exports a default function but that doesn't do anything...)

**Comment by andycarle** [comment]
You can exclude setup/network in your manifest in the same way you would exclude any other inherited module with the ~ directive. See https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/tools/manifest.md#match

**Reply by andycarle** [reply, marked as answer]
To the second half of your question, modules with module specifiers starting with `setup/` are treated in the way you desire.

**Reply by tve** [reply]
Ah, interesting, I had forgotten about the "~"...

**Reply by andycarle** [reply]
The magic happens around here: https://github.com/Moddable-OpenSource/moddable/blob/c58c6e9910ae270fd8f7c7a9758de367331e94f8/xs/platforms/mc/xsHosts.c#L570

**Reply by phoddie** [reply]
@andycarle is, of course, correct about how to remove the setup/network entirely. Another approach is imply not to configure it. If `ssid` is not set or is set to an empty string, the module won't do anything. Then you can use whatever approach you like to configure and mange your Wi-Fi connection.

You can create your own setup modules. As noted in the [Setup documentation](https://www.moddable.com/documentation/base/setup), any module specifier that begins with "setup/" will be run during the set-up phase.
